### DISCUSSION TITLE
PR comparison using outdated file

### DISCUSSION THREAD

**Discussion by augierakow** [community/community#73084] [Pull Requests]
### Select Topic Area

Bug - Web ui's PR request feature is using outdated version of `main`

### Body

I'm trying to use the web ui to create a new Pull Request, so that I can compare my 'branch' against my 'main'.  I expect the PR comparison to use the CURRENT version of each.  But actually, it's comparing the CURRENT 'branch' against an OUTDATED 'main'.  

This is unexpected for two reasons:
1) When I view 'main' in the web ui's Code tab, it correctly shows the CURRENT 'main'. Only the PR comparison is using the OUTDATED 'main'.
2) When I run 'git diff branch..main' from the command line, it correctly diffs against the CURRENT 'main'. Again, only the web ui's PR comparison is using the OUTDATED 'main'.

Seems like a bug causing only the PR tab to use the OUTDATED main.  

Screenshots:
1. Web ui 'Code' tab correctly showing CURRENT main
2. Web ui 'Pull Request' tab incorrectly diffing against OUTDATED main
3. Command line correctly diffing against CURRENT main

Please note, for testing purposes:
a) The CURRENT 'main' includes the string "TEST MAIN BRANCH" at line 1
b) The OUTDATED 'main' lacks that string in line 1
c) The CURRENT 'branch' includes the string "TEST FAFO BRANCH" at line 1.

<img width="856" alt="image" src="https://github.com/community/community/assets/11730069/6e4ba16d-ebcc-40a7-b327-a08099d9136f">

<img width="878" alt="image" src="https://github.com/community/community/assets/11730069/6244cce5-6a5b-4e56-ad9f-ed74c9b6e1af">

<img width="858" alt="image" src="https://github.com/community/community/assets/11730069/52f3fd82-5cb6-4c65-823e-9b17dba0ed8e">

How do I fix this?

**Comment by glerendegui** [comment]
Same here. This behaviour makes PRs totally unuseful for us

**Comment by ktkennychow** [comment]
How is this bug is still not solved is beyond me, Hi Microsoft?
<img width="1322" alt="Screenshot 2025-03-12 at 10 52 04" src="https://github.com/user-attachments/assets/51ea650a-404f-4e52-bd35-e32ee0b4101e" />
<img width="367" alt="Screenshot 2025-03-12 at 10 50 48" src="https://github.com/user-attachments/assets/c7da96a4-82c3-4f29-8aa9-f45214a49ea2" />
The Comparing Changes part of Github PR, should simply be `git merge` but opened for team members to check conflicts and resolve them.
